docs/source/design/user_messaging.rst

```diff
@@ -0,0 +1,110 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_user_messaging:
+
+*******************************************************
+User messaging: result records vs exceptions vs logging
+*******************************************************
+
+.. topic:: Specification scope and status
+
+   This specification provides a partial overview of the current implementation.
+
+Motivation
+==========
+
+This specification delineates the applicable contexts for using
+:ref:`result records <chap_design_result_records>`, :ref:`exceptions <chap_design_exception_handling>`,
+:ref:`progress reporting <chap_design_progress_reporting>`, specific :ref:`log levels <chap_design_log_levels>`,
+or other types of user messaging processes.
+
+
+Specification
+=============
+
+Result records
+--------------
+
+**Result records are the only return value format** for all DataLad interfaces.
+
+Constrasting with classic Python interfaces that return specific non-annotated values,
```

Comment:
```suggestion
Contrasting with classic Python interfaces that return specific non-annotated values,
```

Reply:
corrected in [ee1c43b](https://github.com/datalad/datalad/pull/7310/commits/ee1c43bd67650db725a78ab4f293e181e064e807)